src/app/components/wrapper-components/rules-configuration/rules-configuration.component.ts

```diff
@@ -0,0 +1,221 @@
+            const parameterNames: string[] = Object.keys(config);
+            for (const parameterName of parameterNames) {
+                if (this.isInvalid(parameterName)) {
+                    // This inform the parent component that an invalid update has been done
+                    this.updateCallback.emit(MGPOptional.empty());
+                    return; // In order not to send update when form is invalid
+                }
+                rulesConfig[parameterName] = this.rulesConfigForm.controls[parameterName].value;
+            }
+            this.setConfigDemo(rulesConfig);
+            this.updateCallback.emit(MGPOptional.of(rulesConfig));
+        }
+    }
+
+    public isNumber(field: string): boolean {
+        const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+        const value: ConfigDescriptionType = config[field];
+        return typeof value === 'number';
+    }
+
+    public isBoolean(field: string): boolean {
+        const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+        const value: ConfigDescriptionType = config[field];
+        return typeof value === 'boolean';
+    }
+
+    public isInvalid(field: string): boolean {
+        const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+        const value: ConfigDescriptionType = config[field];
+        if (typeof value === 'number') {
+            const fieldValue: number = this.rulesConfigForm.controls[field].value;
+            const validity: MGPValidation = this.rulesConfigDescription.validator[field](fieldValue);
+            return validity.isFailure();
+        } else {
+            Utils.expectToBe(typeof value, 'boolean');
+            // Angular make thoses controls invalid when they are boolean, not sure why
+            return false; // So we return false cause they cannot be invalid, checked or not, its always valid
+        }
+    }
+
+    public getErrorMessage(field: string): string {
+        const fieldValue: number | null = this.rulesConfigForm.controls[field].value;
+        const validity: MGPValidation = this.rulesConfigDescription.validator[field](fieldValue);
+        return validity.getReason();
+    }
+
+    public getFields(): string[] {
+        const config: RulesConfig = this.rulesConfigDescription.getDefaultConfig().config;
+        return Object.keys(config);
+    }
+
+    public onChange(event: Event): void {
+        const select: HTMLSelectElement = event.target as HTMLSelectElement;
+        this.setChosenConfig(select.value);
+    }
+
+    private setChosenConfig(configName: string, emit: boolean = true): void {
+        this.chosenConfig = configName;
+        let config: RulesConfig;
+        if (this.chosenConfig === 'Custom') {
+            config = this.rulesConfigDescription.getDefaultConfig().config;
+            this.generateForm(config, this.userIsCreator);
```

Comment:
j'sais plus si c'était noté, mais : bien que le form est pas éditable par le non-créateur, le non-créateur peut afficher le menu déroulant de configs !